tests/measurements/test_measurements_legacy.py

```diff
@@ -107,6 +107,9 @@ class MyMeasurement(StateMeasurement):
             def process_state(self, state, wire_order):
                 return qml.math.sum(state)
 
+            def process_density_matrix(self, density_matrix, wire_order):
```

Comment:
I took a look at `DefaultMixed` and I don't think it actually ever uses `process_density_matrix`. It inherits from `QubitDevice` and returns the result of a measurement from this:
https://github.com/PennyLaneAI/pennylane/blob/c2dcfaac2c3d5fd08e0065337274213b2f8f01aa/pennylane/devices/_qubit_device.py#L574
where `self.state` of a `DefaultMixed` device is a density matrix. This is actually a bug, because if you consider something like this:
```python
class MyMeasurement(StateMeasurement):
    
    def process_state(self, state, wire_order):
        with qml.queuing.QueuingManager.stop_recording():
            return qml.expval(qml.Z(0)).process_state(state, wire_order=wire_order)

dev = qml.device("default.mixed", wires=2)

@qml.qnode(dev)
def circuit():
    return MyMeasurement()
```
```pycon
>>> circuit()
array([1., 0., 0., 0.])
```
You get an array as the expected value, which is clearly wrong.

This is not within the scope of this PR, and I assume it'll be fixed in [this epic](https://app.shortcut.com/xanaduai/epic/73315?group_by=none&vc_group_by=day&ct_workflow=all&cf_workflow=500000005) so I'm not too worried, but it's something to keep in mind.

Reply:
Added this to the epic as another concrete reason to port to the new device API and stop relying on the confusing spaghetti of assumptions that is `QubitDevice`.